python/lsst/meas/extensions/multiprofit/fit_coadd_multiband.py

```diff
@@ -75,58 +127,60 @@ def get_source_observation(self, source) -> g2f.Observation:
             bitmask |= bitval
         mask = ((mask.array & bitmask) != 0) & (spans != 0)
         mask = ~mask
+
+        is_deblended_child = source["parent"] != 0
+
         img, _, sigma_inv = get_spanned_image(
             exposure=self.exposure,
+            footprint=footprint if is_deblended_child else None,
             bbox=bbox,
             spans=spans,
             get_sig_inv=True,
         )
+        x_min_bbox, y_min_bbox = bbox.beginX, bbox.beginY
+        # Crop to tighter box for deblended model if edges are unusable
+        # ... this rarely ever seems to happen though
+        if is_deblended_child:
+            coords = np.argwhere(np.isfinite(img) & (sigma_inv > 0) & np.isfinite(sigma_inv))
+            x_min, y_min = coords.min(axis=0)
+            x_max, y_max = coords.max(axis=0)
+            x_max += 1
+            y_max += 1
+
+            if (x_min > 0) or (y_min > 0) or (x_max < img.shape[0]) or (y_max < img.shape[1]):
```

Comment:
I'm not sure why you need this check. Shouldn't you always check that the source is inside the bbox and adjust it if not?

Reply:
x/y_min/max here are indices of columns with finite data. It's checking if there are entire rows/columns at the edge of the bbox with bad data that can be cropped, since they'd have no impact on the fit anyway. This happens very rarely so I was considering dropping it entirely.